### PR TITLE
XW-3979 | prevent indexing 404 and private wikis in LI

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3979
@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 1]
```diff
@@ -123,7 +122,7 @@ function () {
 	}
 
 	private static function addLiftIgniterMetadata( OutputPage $outputPage ) {
-		global $wgDevelEnvironment, $wgLanguageCode, $wgStagingEnvironment;
+		global $wgDevelEnvironment, $wgLanguageCode, $wgStagingEnvironment, $wgIsPrivateWiki, $wgCityId;;
```
Comment (WikiaTech) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: MINOR Remove this empty statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 2]
```diff
@@ -135,8 +134,9 @@ private static function addLiftIgniterMetadata( OutputPage $outputPage ) {
 
 		$metaData['language'] = $wgLanguageCode;
 		$isProduction = empty( $wgDevelEnvironment ) && empty( $wgStagingEnvironment );
+		$isPrivateWiki = WikiFactory::isWikiPrivate( $wgCityId ) || $wgIsPrivateWiki;
```
Comment (hakubo) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: why both? :) ❓

File: extensions/wikia/Recirculation/RecirculationHooks.class.php [hunk 3]
```diff
@@ -135,8 +134,9 @@ private static function addLiftIgniterMetadata( OutputPage $outputPage ) {
 
 		$metaData['language'] = $wgLanguageCode;
 		$isProduction = empty( $wgDevelEnvironment ) && empty( $wgStagingEnvironment );
```
Comment (rybmat) [inline extensions/wikia/Recirculation/RecirculationHooks.class.php]: should we consider staging as production here?

File: extensions/wikia/Recirculation/RecirculationHooks.class.php (hunk 2 above)
Comment (latata) [reply extensions/wikia/Recirculation/RecirculationHooks.class.php]: `WikiFactory::isWikiPrivate` checks for access permissions set in `wgGroupPermissionsLocal` content.
`$wgIsPrivateWiki` is just a boolean variable

**Review by rybmat — APPROVED** [review]

**Review by hakubo — APPROVED** [review]

**Pull request merged**